### PR TITLE
add coroutines-play-services as a transitive dep to firebase-common-ktx

### PR DESCRIPTION
**Pull request by thatfiredev**
_(Googlers see [go/suspend-ktx-firebase-android](http://go/suspend-ktx-firebase-android))_

### CODE REVIEW

**Comment by thatfiredev** [comment]
I'm slightly concerned with the increased size shown in the bot report above. Seems like adding the library as a transitive dependency adds **~390kB** to every ktx module - to me it is a big bump for a small addition.
I'll check if we can expose only the `.await()` function instead of the whole library to see if it makes any difference in terms of  size.

**Comment by thatfiredev** [comment]
I'm holding this until #3974 gets merged. I think it would be better if we release with the latest version of Kotlin + latest version of kotlinx-coroutines-play-services .

File: firebase-common/ktx/src/test/kotlin/com/google/firebase/ktx/Tests.kt
Comment (vkryachko) [inline firebase-common/ktx/src/test/kotlin/com/google/firebase/ktx/Tests.kt:141]: curious: what's blocking us from supporting that?

Comment (thatfiredev) [reply firebase-common/ktx/src/test/kotlin/com/google/firebase/ktx/Tests.kt]: @vkryachko you were a few minutes ahead of me, hehe.
Using Coroutines 1.6 was throwing errors during compilation because we were using an older version of Kotlin. Now that we're using Kotlin 1.7 it works fine.

Comment (thatfiredev) [reply firebase-common/ktx/src/test/kotlin/com/google/firebase/ktx/Tests.kt]: But it seems like none of our APIs support passing a CancellationToken, so I'm not adding a test for that.

Comment (vkryachko) [reply firebase-common/ktx/src/test/kotlin/com/google/firebase/ktx/Tests.kt]: yeah, my hope is that tasks become cancellable directly, that way we could get proper cancellation without changes required in all of our API surfaces

**Review by vkryachko — APPROVED** [review]

**Comment by thatfiredev** [comment]
@vkryachko Can I merge this?

Also since it bumps firebase-common, does it mean we'll have to release all SDKs? Or only the ones that have methods returning Tasks?
But since it's a fireconf launch, I guess releasing them all wouldn't be so bad.

**Comment by vkryachko** [comment]
Sure, go ahead and merge. Yeah, we will have to release everyone with tasks

**Comment by thatfiredev** [comment]
@vkryachko will do. Thanks!

**Comment by thatfiredev** [comment]
@vkryachko GitHub won't let me merge this, it says "Required statuses must pass before merging", I'm not sure why.

**Comment by thatfiredev** [comment]
@vkryachko I checked the logs and noticed that the failing check was the flaky test which got fixed recently. I just had to sync with master, thanks for pointing me in the right direction.

**Pull request merged**
